sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -171,6 +171,11 @@ case class AttributeReference(
     val exprId: ExprId = NamedExpression.newExprId,
     val qualifiers: Seq[String] = Nil) extends Attribute with trees.LeafNode[Expression] {
 
+  /**
+   * Returns true iff the expression id is the same for both attributes.
```

Comment:
`iff` is [a pretty standard shortening](http://en.wikipedia.org/wiki/If_and_only_if) and is used [throughout the spark code base already](https://github.com/apache/spark/search?utf8=%E2%9C%93&q=iff).